bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/AsterismConverters.scala

```diff
@@ -0,0 +1,108 @@
+package edu.gemini.spModel.gemini.ghost
+
+import edu.gemini.shared.util.immutable.ImList
+import edu.gemini.spModel.core.{Coordinates, SiderealTarget}
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.target.env.{TargetEnvironment, UserTarget}
+
+import scalaz._
+import Scalaz._
+
+/** Allow conversions between target environments containing the different GHOST asterism types.
+  * Coordinates are immutable, but SPTargets are mutable, so we must clone them.
+  */
+object AsterismConverters {
+  import GhostAsterism._
+  import GhostAsterism.GhostStandardResTargets._
+
+  type BasePosition = Option[Coordinates]
+  type SkyPosition  = Option[Coordinates]
+
+  sealed trait AsterismConverter {
+    def name: String
+    def convert(env: TargetEnvironment): String \/ TargetEnvironment
+  }
+
+  sealed trait GhostAsterismConverter extends AsterismConverter {
+    override def convert(env: TargetEnvironment): String \/ TargetEnvironment = env.getAsterism match {
+        case StandardResolution(SingleTarget(t), b)     => creator(env, t,  None,    None,   b).right
+        case StandardResolution(DualTarget(t1, t2), b)  => creator(env, t1, t2.some, None,   b).right
+        case StandardResolution(TargetPlusSky(t, s), b) => creator(env, t,  None,    s.some, b).right
+        case StandardResolution(SkyPlusTarget(s, t), b) => creator(env, t,  None,    s.some, b).right
+        case HighResolution(t, s, b)                    => creator(env, t,  None,    s,      b).right
+        case _                                          => s"Could not convert to $name".left
+      }
+
+    protected def creator(env: TargetEnvironment, t1: GhostTarget, t2: Option[GhostTarget], s: SkyPosition, b: BasePosition): TargetEnvironment
```

Comment:
This is just a method that takes the targets / coordinates common to all asterism types so that we can more easily extract parameters from one asterism type and turn it into another without having to have n^2 cases, were n is the number of asterism types.

Reply:
😊 
Good idea.